index.js

```diff
@@ -99,12 +99,32 @@ module.exports = {
 
     // We shouldn't enforce filename casing to always be the same.  e.g. We may
     // prefer kebab-case for bin files, PascalCase for React components, etc.
+    // https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/docs/rules/filename-case.md
     'unicorn/filename-case': 'off',
 
     // Uses safe-regex to disallow potentially catastrophic exponential-time
     // regular expressions.
+    // https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/docs/rules/no-unsafe-regex.md
     'unicorn/no-unsafe-regex': 'error',
 
+    // This rule is onerous in React functional components, has been debated
+    // by the community and proven as an unnecessary optimization
+    // https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/docs/rules/consistent-function-scoping.md
+    'unicorn/consistent-function-scoping': 'off',
+
+    // This rule is incompatible with React
```

Comment:
The rule summary is
>Disallow the use of the null literal, to encourage using undefined instead.

This comment might be more useful phrased something like
>React makes specific differentiation between `null` and `undefined`

Reply:
It's not just React. There is a tangible difference between null and undefined and people who code in Javascript should know the difference. Rules like this are training wheels for people who don't know the difference. We should strive for knowledge, not ignorance.

Reply:
All of my code comments were intended as rough drafts. We can change them to whatever the consensus decides they should be.